### PR TITLE
Maintain uniques on models whose id is set later (fixes #13)

### LINKED ISSUE #13: Maintain uniques when ID is set retroactively
This would be super useful in cases where multiple views have a new model that gets saved.  It is difficult to update those models to the uniquemodel version without this functionality.  @jfschwarz 's commit here seems like it would do the trick.

https://github.com/jfschwarz/backbone.uniquemodel/commit/e922f72b2fc534c4eff96236427a6cde5b8f5db1

### PR DESCRIPTION
**Pull request by benvinegar**

### CODE REVIEW

**Comment by paulfalgout** [comment]
:clap: :+1:

File: backbone.uniquemodel.js [hunk 1]
```diff
@@ -117,6 +117,15 @@
     newModel: function (attrs, options) {
       var instance = new this.Model(attrs, options);
 
+      if (!instance.id) {
+        // If this model currently has no id, but gains one later (e.g. via sync),
+        // then add it to the list of tracked instances
+        instance.once('change:' + instance.idAttribute, function () {
```
Comment (BYK) [inline backbone.uniquemodel.js:123]: Are you sure you cannot simply use `change:id`?

File: backbone.uniquemodel.js [hunk 2]
```diff
@@ -117,6 +117,15 @@
...
+          if (!this.instances[instance.id])
+            this.instances[instance.id] = instance;
```
Comment (BYK) [inline backbone.uniquemodel.js:125]: Upcoming: merge with the existing instance if there is one?

**Comment by BYK** [comment]
LGTM, that said GitHub says I cannot merge this automatically. Would you mind merging master back into this or rebasing onto master?

File: backbone.uniquemodel.js (hunk 1 above)
Comment (benvinegar) [inline backbone.uniquemodel.js:123]: Yes. Just verified in the console:

``` javascript
var User = Backbone.Model.extend({idAttribute: 'username'});
var user = new User({username: 'johndoe'});

user.on('change:id', function () { console.log('id fired'); })
user.on('change:username', function () { console.log('username fired'); })
user.set('username', 'janedoe')
// username fired
```

File: backbone.uniquemodel.js (hunk 2 above)
Comment (benvinegar) [inline backbone.uniquemodel.js:125]: Can you give me an example of this scenario? Not sure I follow.

File: backbone.uniquemodel.js (hunk 1 above)
Comment (BYK) [inline backbone.uniquemodel.js:123]: Eh, bad Backbone :(

File: backbone.uniquemodel.js (hunk 2 above)
Comment (BYK) [inline backbone.uniquemodel.js:125]: The thing that we're already doing here: https://github.com/disqus/backbone.uniquemodel/blob/master/backbone.uniquemodel.js#L209-L210

lol, sorry my bad. Just ignore :)

**Comment by paulfalgout** [comment]
Any news on a rebase?

**Comment by BYK** [comment]
Welp, too late @benvinegar: #17.

**Pull request closed without merging**